src/Sentry.Unity/SentryOptionsUtility.cs

```diff
@@ -1,42 +1,62 @@
 using Sentry.Extensibility;
 using Sentry.Unity.Integrations;
+using UnityEngine;
 
 namespace Sentry.Unity
 {
     internal static class SentryOptionsUtility
     {
         public static void SetDefaults(SentryUnityOptions options, IApplication? application = null)
         {
+            options.TryAttachLogger();
+
             application ??= ApplicationAdapter.Instance;
 
             SetRelease(options, application);
             SetEnvironment(options, application);
             SetCacheDirectoryPath(options, application);
         }
 
+        public static void LogOptions(SentryUnityOptions options)
+        {
+            var logger = options.DiagnosticLogger;
+            if (logger == null)
+            {
+                return;
+            }
+
+            LogOption(logger, "Release", options.Release);
+            LogOption(logger, "Environment", options.Environment);
+
+            if (options.CacheDirectoryPath == null)
+            {
+                logger.Log(SentryLevel.Debug, "Offline Caching disabled.");
+            }
+            else
+            {
+                LogOption(logger, "Cache Directory", options.CacheDirectoryPath);
+            }
+        }
+
         private static void SetRelease(SentryUnityOptions options, IApplication application)
         {
             options.Release ??= application.ProductName is string productName
                 && !string.IsNullOrWhiteSpace(productName)
                     ? $"{productName}@{application.Version}"
                     : $"{application.Version}";
-
-            Log(options.DiagnosticLogger, "Release", options.Release);
         }
 
         private static void SetEnvironment(SentryUnityOptions options, IApplication application)
         {
             options.Environment ??= application.IsEditor ? "editor" : "production";
-            Log(options.DiagnosticLogger, "Environment", options.Environment);
         }
 
         private static void SetCacheDirectoryPath(SentryUnityOptions options, IApplication application)
         {
             options.CacheDirectoryPath ??= application.PersistentDataPath;
-            Log(options.DiagnosticLogger, "Cache Directory", options.CacheDirectoryPath);
         }
 
-        private static void Log(IDiagnosticLogger? logger, string option, object value)
-            => logger?.Log(SentryLevel.Debug, "Setting Sentry {0} to: {1}", null, option, value);
+        private static void LogOption(IDiagnosticLogger logger, string option, object value)
```

Comment:
`object value` will box (allocate) any value type. Ideally we'd take `<T>` and first check if Logger.IsEnabled for that level before we allocate memory (otherwise it's just a waste).

The .NET SDK has proper extension methods which was made public in the .NET SDK 3.4.0 so if we bump the sub module here we can use it.

No more `logger.Log(level, ..`. We just use `LogDebug(..` at that point

https://github.com/getsentry/sentry-dotnet/blob/1095abca85eeefaac7170e7dcf482ffc052d53c9/src/Sentry/Extensibility/DiagnosticLoggerExtensions.cs#L11